remote-dependency-templates/README.md

```diff
@@ -1,11 +1,87 @@
-## Usage
+# Using dependency templates
 - List dependency `<id>` as a remote dependency in `application.yml`, with one of the supported types
   - Populate required `template-config` fields (defined in the README for each type)
 - List any dependency-specific service data under `dependency-data.<id>` in a service's `service.yml`
 
-## Terraform variables //TODO
+# Building dependency templates
+- Create a directory with a unique name. This will be the dependency's `type`.
+- The directory should contain the following three files: requirements.yml, dependency.tf and README.md. Each is explained in detail below. (see existing templates for example)
 
-## Notes
-The following are fields that exosphere passes to every dependency:
-- `app-name`: Name of the application as defined in `application.yml`
+## dependency.tf
+A terraform file that contains the public interfacing terraform modules for a dependency. The file should be built
+ as a [Mustache template](https://github.com/hoisie/mustache) which will be filled in with the user defined template-config.
+ You can ensure the user passes in all the required fields using the requirements.yml file.
+Example:
+```tf
+# dependency.tf
+module "example_module" {
+  version = "{{version}}"
+}
+```
+```yml
+# application.yml
+remote:
+  dependencies:
+    <dependency-id>:
+      type: example
+      template-config:
+        version: 0.0.1
+```
+```
+# rendered dependency in user's main.tf
+# dependency.tf
+module "example_module" {
+  version = "0.0.1"
+}
+```
 
+Submodules should be sourced using git URLs: https://www.terraform.io/docs/modules/sources.html#github (see below for more details).
+The following fields are automatically rendered into the mustache template for each dependency:
+- `terraformCommitHash`: the git commit hash of the latest changes to any terraform files of this repository
+
+## Terraform modules
+Any submodules of the main `dependency.tf` should be defined in `#{dependencyType}/modules`, and should be referenced from `dependency.tf` using git URLs.
```

Comment:
I think this section should be updated to

```
## modules
An optional directory containing submodules of `dependency.tf`.
```

I don't think we need to mention git URLs again since that is already mentioned in `dependency.tf`.

Reply:
Why change it from `Terraform modules` -> `modules`?

Reply:
The titles are the file / directory name